Copyright 2014 The Chromium Authors. All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are
met:

    * Redistributions of source code must retain the above copyright
notice, this list of conditions and the following disclaimer.
    * Redistributions in binary form must reproduce the above
copyright notice, this list of conditions and the following disclaimer
in the documentation and/or other materials provided with the
distribution.
    * Neither the name of Google Inc. nor the names of its
contributors may be used to endorse or promote products derived from
this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
ATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

Date: Tue, 19 Dec 2017 10:30:06 -0800

From: da59d58c03b0e58c5cdf570539324120ef6314a6 Mon Sep 17 00:00:00 2001

Subject: [PATCH] WKIT-1041: WPEWebProcess crash in OpenCDM:update()

	check for mediakeysessionupdate failure

Source: COMCAST

Upstream-Status: Pending

Signed-off-by: Gurdal Oruklu <Gurdal_Oruklu2@cable.comcast.com>
---
 src/browser/wpe/opencdm/open_cdm.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/browser/wpe/opencdm/open_cdm.cpp b/src/browser/wpe/opencdm/open_cdm.cpp
index a5660d1..b0f0e75 100644
--- a/src/browser/wpe/opencdm/open_cdm.cpp
+++ b/src/browser/wpe/opencdm/open_cdm.cpp
@@ -171,9 +171,15 @@ int OpenCdm::Update(unsigned char* pbResponse, int cbResponse, std::string& resp
     cout << pbResponse[i] <<"";
   CDM_DLOG() << "\nEnd";
   CDM_DLOG() << "Update session with info from server.";
-  platform_->MediaKeySessionUpdate((uint8_t*)pbResponse, cbResponse, m_session_id.session_id, m_session_id.session_id_len);
+  MediaKeySessionUpdateResponse status = platform_->MediaKeySessionUpdate((uint8_t*)pbResponse, cbResponse, m_session_id.session_id, m_session_id.session_id_len);
   CDM_DLOG() << "Update session with info from server complete.";
 
+  if (status.platform_response !=  PLATFORM_CALL_SUCCESS) {
+    m_eState = KEY_SESSION_ERROR;
+    CDM_DLOG() << "MediaKeySessionUpdate() failed!";
+    return 1;
+  }
+
   while (m_eState == KEY_SESSION_WAITING_FOR_LICENSE) {
     CDM_DLOG() << "Waiting for license update status!" << "\n";
     fflush(stdout);
-- 
2.14.2

